In [22]:
import pandas as pd
import os
import glob2
import re

In [23]:
def extract_last_number(my_string):
    # Use regex to find a sequence of digits at the end of the string (1-3 digits)
    match = re.search(r'(\d{1,3})$', my_string)
    if match:
        return int(match.group(1))
    else:
        return None  # No digits found

csvs = glob2.glob('/Users/kate_fieseler/PycharmProjects/CHIKV-Mac-syndirella-run/csv_output/csvs_aug5_0952/*.csv')
rows = []
final_df = None
for csv in csvs:
    df = pd.read_csv(csv)
    # add date and time and input
    date = os.path.basename(csv).split('_')[3]
    time = os.path.basename(csv).split('_')[4].split('.')[0]
    input_num = extract_last_number(os.path.basename(csv).split('_')[1])
    df['date'] = date
    df['time'] = time
    df['input_num'] = input_num
    if final_df is None:
        final_df = df
    else:
        # Align the columns of output_df and row_df
        combined_columns = final_df.columns.union(df.columns)
        final_df = final_df.reindex(columns=combined_columns)
        df = df.reindex(columns=combined_columns)
        final_df = pd.concat([final_df, df], ignore_index=True)

In [24]:
df = final_df
df

1_r1_smiles                      1_r2_smiles  \
0         NC1CCC(=O)NC1       NC(=O)c1ccc(C(=O)O)c(Cl)c1   
1                   NaN                              NaN   
2                   NaN                              NaN   
3                   NaN                              NaN   
4             Nc1ccccc1  COC(=O)c1cccc2c(C(=O)O)c[nH]c12   
..                  ...                              ...   
345  CNCc1cccc(=O)[nH]1               CCOC(=O)c1cn[nH]c1   
346  Cc1ccncc1NC(=O)CCl            Cc1[nH][nH]c(=O)c1CCO   
347                 NaN                              NaN   
348                 NaN                              NaN   
349                 NaN                              NaN   

                               1_reaction  \
0                               Amidation   
1                                     NaN   
2                                     NaN   
3                                     NaN   
4                               Amidation   
..                                    ...   
345                       Ester_amidation   
346  Nucleophilic_substitution_with_amine   
347                                   NaN   
348                                   NaN   
349                                   NaN   

                                 2_r1_smiles 2_r2_smiles       2_reaction  \
0                                        NaN         NaN              NaN   
1                                        NaN         NaN              NaN   
2                                        NaN         NaN              NaN   
3                                        NaN         NaN              NaN   
4    COC(=O)c1cccc2c(C(=O)Nc3ccccc3)c[nH]c12          CN  Ester_amidation   
..                                       ...         ...              ...   
345                                      NaN         NaN              NaN   
346                                      NaN         NaN              NaN   
347                                      NaN         NaN              NaN   
348                                      NaN         NaN              NaN   
349                                      NaN         NaN              NaN   

    3_r1_smiles 3_reaction 4_r1_smiles 4_r2_smiles  ... input_num num_placed  \
0           NaN        NaN         NaN         NaN  ...        44      284.0   
1           NaN        NaN         NaN         NaN  ...        30        NaN   
2           NaN        NaN         NaN         NaN  ...        73        NaN   
3           NaN        NaN         NaN         NaN  ...        18        NaN   
4           NaN        NaN         NaN         NaN  ...       277    10000.0   
..          ...        ...         ...         ...  ...       ...        ...   
345         NaN        NaN         NaN         NaN  ...        29     2217.0   
346         NaN        NaN         NaN         NaN  ...        58       44.0   
347         NaN        NaN         NaN         NaN  ...       320        NaN   
348         NaN        NaN         NaN         NaN  ...       165        NaN   
349         NaN        NaN         NaN         NaN  ...        12        NaN   

    num_successful route_uuid  \
0            173.0     32fgo7   
1              NaN        NaN   
2              NaN        NaN   
3              NaN        NaN   
4           9780.0     6PQWWH   
..             ...        ...   
345         1356.0     ZzGYCp   
346           31.0     STWk5f   
347            NaN        NaN   
348            NaN        NaN   
349            NaN        NaN   

                                                smiles  \
0          NC(=O)c1ccc(C(=O)N[C@@H]2CCC(=O)NC2)c(Cl)c1   
1    CN(C(=O)c1cn[nH]c1)[C@H]1C[C@H](O)NC[C@@H]1NC(...   
2                    Cc1cc2c(c(C)c1-c1cc[nH]n1)N(C)CO2   
3              C[C@H](c1ccc(N)nc1)c1nc(Cn2ccccc2=O)no1   
4              CNC(=O)c1cccc2c(C(=O)Nc3ccccc3)c[nH]c12   
..                                                 ...   
345                CN(Cc1cccc(=O)[nH]1)C(=O)c1cn[nH]c1   
346

In [25]:
df.compound_set.value_counts()

compound_set
CHIKVMac-DLS-DF003    2
CHIKVMac-DLS-DF002    2
cx1371a-cx0300a       1
cx0399b-cx0270a       1
cx0892a-cx0281a       1
                     ..
cx0543a-cx0316a       1
CHIKVMac-DLS-JA005    1
CHIKVMac-DLS-JA006    1
cx1116a-cx0477a       1
CHIKVMac-DLS-DF004    1
Name: count, Length: 348, dtype: int64

In [30]:
df.input_num.value_counts()

input_num
11     2
10     2
44     1
270    1
20     1
      ..
146    1
4      1
5      1
95     1
12     1
Name: count, Length: 348, dtype: int64

In [27]:
df[df['input_num']==10]

1_r1_smiles                1_r2_smiles 1_reaction 2_r1_smiles  \
310  Nc1cnc2[nH]ccc2c1  Cc1[nH][nH]c(=O)c1CC(=O)O  Amidation         NaN   
311  Nc1cnc2[nH]ccc2c1  Cc1[nH][nH]c(=O)c1CC(=O)O  Amidation         NaN   

    2_r2_smiles 2_reaction 3_r1_smiles 3_reaction 4_r1_smiles 4_r2_smiles  \
310         NaN        NaN         NaN        NaN         NaN         NaN   
311         NaN        NaN         NaN        NaN         NaN         NaN   

     ... input_num num_placed num_successful route_uuid  \
310  ...        10       96.0           81.0     Ndgh2a   
311  ...        10       96.0           77.0     GDNfCh   

                                        smiles  \
310  Cc1[nH][nH]c(=O)c1CC(=O)Nc1cnc2[nH]ccc2c1   
311  Cc1[nH][nH]c(=O)c1CC(=O)Nc1cnc2[nH]ccc2c1   

                                              template  time  \
310  /opt/xchem-fragalysis-2/kfieseler/CHIKV-Mac-sy...  1738   
311  /opt/xchem-fragalysis-2/kfieseler/CHIKV-Mac-sy...  1738   

                                              to_hippo  \
310  /opt/xchem-fragalysis-2/kfieseler/CHIKV-Mac//O...   
311  /opt/xchem-fragalysis-2/kfieseler/CHIKV-Mac//O...   

     total_num_products_enumstereo  total_num_unique_products  
310                          129.0                      126.0  
311                          129.0                      126.0  

[2 rows x 28 columns]

In [28]:
df[df['input_num'] == 11]

1_r1_smiles 1_r2_smiles 1_reaction 2_r1_smiles 2_r2_smiles 2_reaction  \
108         NaN         NaN        NaN         NaN         NaN        NaN   
109         NaN         NaN        NaN         NaN         NaN        NaN   

    3_r1_smiles 3_reaction 4_r1_smiles 4_r2_smiles  ... input_num num_placed  \
108         NaN        NaN         NaN         NaN  ...        11        NaN   
109         NaN        NaN         NaN         NaN  ...        11        NaN   

    num_successful route_uuid                            smiles  \
108            NaN        NaN  Nc1ccc(-c2nc(Cn3ccccc3=O)no2)cn1   
109            NaN        NaN  Nc1ccc(-c2nc(Cn3ccccc3=O)no2)cn1   

                                              template  time to_hippo  \
108  /opt/xchem-fragalysis-2/kfieseler/CHIKV-Mac-sy...  1731      NaN   
109  /opt/xchem-fragalysis-2/kfieseler/CHIKV-Mac-sy...  1731      NaN   

     total_num_products_enumstereo  total_num_unique_products  
108                            NaN                        NaN  
109                            NaN                        NaN  

[2 rows x 28 columns]

In [32]:
# remove input num duplicates
df = df.drop_duplicates(subset=['input_num'])
len(df)

348

In [33]:
# what ints from 0 to 353 are not included
missing = set(range(354)) - set(list(df.input_num))
missing

{40, 87, 283, 302, 304, 329}

In [34]:
# start these
with open('/Users/kate_fieseler/PycharmProjects/CHIKV-Mac-syndirella-run/jobs_6_missing.txt', 'w') as f:
    for i in missing:
        f.write(f'job{i}.sh\n')

Only jobs 40 and 283 had OOM errors, therefore increasing memory to 30GB and restarting. 

In [39]:
# how many scaffolds actually had placements
placed = df[df['num_placed']>0]
placed

1_r1_smiles                      1_r2_smiles  \
0          NC1CCC(=O)NC1       NC(=O)c1ccc(C(=O)O)c(Cl)c1   
4              Nc1ccccc1  COC(=O)c1cccc2c(C(=O)O)c[nH]c12   
8    O=C(O)COc1cccc(F)c1                   CNCc1cccc(O)n1   
10        CNCc1ccc(O)nc1                CCOc1ncccc1C(=O)O   
14            NC1CCCC1CO            Clc1c(I)ccc2[nH]cnc12   
..                   ...                              ...   
328     NOc1ccc(Br)cc1Br                       CC(C)B(O)O   
341                 OCCl                Clc1ccc2c(c1)NCO2   
343                 O=CO      CN(C)C(=O)CNc1cc[nH]c(=O)c1   
345   CNCc1cccc(=O)[nH]1               CCOC(=O)c1cn[nH]c1   
346   Cc1ccncc1NC(=O)CCl            Cc1[nH][nH]c(=O)c1CCO   

                               1_reaction  \
0                               Amidation   
4                               Amidation   
8                               Amidation   
10                              Amidation   
14             Buchwald-Hartwig_amination   
..                                    ...   
328               Sp3-sp2_Suzuki_coupling   
341  Nucleophilic_substitution_with_amine   
343                             Amidation   
345                       Ester_amidation   
346  Nucleophilic_substitution_with_amine   

                                 2_r1_smiles      2_r2_smiles  \
0                                        NaN              NaN   
4    COC(=O)c1cccc2c(C(=O)Nc3ccccc3)c[nH]c12               CN   
8                                        NaN              NaN   
10                                       NaN              NaN   
14                                       NaN              NaN   
..                                       ...              ...   
328                                      NaN              NaN   
341                      OCN1COc2ccc(Cl)cc21  OB(O)c1ccn[nH]1   
343                                      NaN              NaN   
345                                      NaN              NaN   
346                                      NaN              NaN   

                  2_reaction 3_r1_smiles 3_reaction 4_r1_smiles 4_r2_smiles  \
0                        NaN         NaN        NaN         NaN         NaN   
4            Ester_amidation         NaN        NaN         NaN         NaN   
8                        NaN         NaN        NaN         NaN         NaN   
10                       NaN         NaN        NaN         NaN         NaN   
14                       NaN         NaN        NaN         NaN         NaN   
..                       ...         ...        ...         ...         ...   
328                      NaN         NaN        NaN         NaN         NaN   
341  Sp2-sp2_Suzuki_coupling         NaN        NaN         NaN         NaN   
343                      NaN         NaN        NaN         NaN         NaN   
345                      NaN         NaN        NaN         NaN         NaN   
346                      NaN         NaN        NaN         NaN         NaN   

     ... input_num num_placed num_successful route_uuid  \
0    ...        44      284.0          173.0     32fgo7   
4    ...       277    10000.0         9780.0     6PQWWH   
8    ...       134     1542.0         1276.0     qehMd6   
10   ...        50     3290.0         1684.0     tTnD23   
14   ...       238     2140.0          477.0     bk89Er   
..   ...       ...        ...            ...        ...   
328  ...       115     1300.0          236.0     vCybEb   
341  ...       300    10000.0           28.0     6ACutS   
343  ...       161      815.0           15.0     4DEaMj   
345  ...        29     2217.0         1356.0     ZzGYCp   
346  ...        58       44.0           31.0     STWk5f   

                                          smiles  \
0    NC(=O)c1ccc(C(=O)N[C@@H]2CCC(=O)NC2)c(Cl)c1   
4        CNC(=O)c1cccc2c(C(=O)Nc3ccccc3)c[nH]c12   
8             CN(Cc1cccc(O)n1)C(=O)COc1cccc(F)c1   
10              CCOc1ncccc1C(=O)N(C)Cc1ccc(O)nc1   
14         OC[C@@H]1CCC[C@H]1Nc1ccc2[nH]

In [40]:
# add up number of placements
sum(placed.num_successful)

107571.0